src/net_processing.cpp

```diff
@@ -5666,7 +5666,9 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                     // especially since we have many peers and some will draw much shorter delays.
                     unsigned int nRelayedTransactions = 0;
                     LOCK(tx_relay->m_bloom_filter_mutex);
-                    while (!vInvTx.empty() && nRelayedTransactions < INVENTORY_BROADCAST_MAX) {
+                    size_t broadcast_max{INVENTORY_BROADCAST_MAX + (tx_relay->m_tx_inventory_to_send.size()/1000)*5};
```

Comment:
Could this increase the number of transactions we're announcing beyond `INVENTORY_MAX_RECENT_RELAY = 3500` so things fall out of `m_recently_announced_invs` (essentially what you said in https://github.com/bitcoin/bitcoin/pull/27602#discussion_r1189250200)?

Maybe we want to increase `INVENTORY_MAX_RECENT_RELAY`, e.g. equal to `MAX_PEER_TX_ANNOUNCEMENTS = 5000` (the max requesting coming from the receiver end if they're Bitcoin Core - imo it could make sense for these to be symmetrical)?

Reply:
The `3500` figure is calculated based on outbound peers, which already have 2.5x the INV rate of inbound peers. An inbound peer would need to be at 65 or 70 txids per INV message to have the same 1-in-a-million chance of a problem, implying a tx rate of 13 or 14 tx/s. To get to a 1-in-100 chance of a problem, you'd need to hit 90-to-95 txids per INV (and sustain that for ~40 INVs, and hit the 1-in-100 chance of sending those 40 INVs in a 2 minute period). Those seem okay to me.

Reply:
Note to others: we may send outbound peers 7 * 2.5 * 60 * 2 ~= 2100 transaction ids per 2 minutes. Depending on random variation it can be more. `UNCONDITIONAL_RELAY_DELAY = 2min;` is the period of time that we have to remember what we sent to a specific peer. For privacy reason we don't want to reveal these transactions to _other_ (spy) peers yet. We do this tracking with a bloom filter `m_recently_announced_invs` that has a 1 in a million chance of messing up. I forgot if it's false positive (we give the spy a transaction we shouldn't have) or negative (we deny them a transaction we just announced).


This change increases the max rate for outbound rate too, so I'm still confused why we don't have to increase the `3500`. This PR increases the maximum rate to outbound peers from  7 * 2.5 = 17.5 tx/s to 1000 / 2 = 500 tx/s (in theory, presumably not in practice), so a 4 second burst would fill it up.

From [below](https://github.com/bitcoin/bitcoin/pull/27610/commits/5b3406094f2679dfb3763de4414257268565b943#r1190574243):

> (Could make sense to limit it to 90 for inbounds and 45 for outbounds as (if sustained for two minutes) those values would hit roughly a 1-in-100 chance of overflowing `INVENTORY_MAX_RECENT_RELAY`)

That would make more sense to me as well, plus:

> static asserts or something would be even better